### ISSUE TITLE
Places - "An error occurred"

### ISSUE THREAD

**Issue by vanantwerp** [nextcloud/photos#1717]
**Describe the bug**
When I click the "Places" menu item (left hand nav) the screen displays "An error occurred" with an exclamation mark symbol. Nothing else appears to happen in the UI. It throws an error in admin logs (below).

**To Reproduce**
1. Go to Photos app
2. Click on Places (left hand nav)
3. Error appears

**Expected behavior**
Unsure what this feature is.

**Screenshots**
![2023-03-28 22_44_59-Photos - Photos - Nextcloud — Mozilla Firefox](https://user-images.githubusercontent.com/17768949/228413253-2bd2821f-28f1-45a6-91a6-8e47e8f776eb.png)

**Client**
**Desktop (please complete the following information):**
 - OS: Win 11
 - Browser: Firefox
 - Version: 111.0.1

**Browser log**
```
500 error for PROPFIND myserver.com/remote.php/dav/photos/user/places/
```

**Server**
- Ubuntu / Docker
- Nextcloud Hub 4 (26.0.0)
- postgres
- nginx
- Photos 2.2.0

**Admin log**
```
[webdav] Error: OC\DB\Exceptions\DbalException: An exception occurred while executing a query: SQLSTATE[42883]: Undefined function: 7 ERROR:  could not identify an equality operator for type json
LINE 1: SELECT DISTINCT "meta"."metadata" FROM "oc_file_metadata" "m...
                        ^ at <<closure>>

 0. /var/www/html/lib/private/DB/QueryBuilder/QueryBuilder.php line 295
    OC\DB\Exceptions\DbalException::wrap(["Doctrine\\DBAL ... "])
 1. /var/www/html/apps/photos/lib/DB/Place/PlaceMapper.php line 62
    OC\DB\QueryBuilder\QueryBuilder->executeQuery()
 2. /var/www/html/apps/photos/lib/Sabre/Place/PlacesHome.php line 95
    OCA\Photos\DB\Place\PlaceMapper->findPlacesForUser("user")
 3. /var/www/html/3rdparty/sabre/dav/lib/DAV/Tree.php line 200
    OCA\Photos\Sabre\Place\PlacesHome->getChildren()
 4. /var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php line 900
    Sabre\DAV\Tree->getChildren("photos/user/places")
 5. /var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php line 982
    Sabre\DAV\Server->generatePathNodes(["Sabre\\DAV\\PropFind"], [["Sabre\\DAV\\P ... ]])
 6. /var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php line 1662
    Sabre\DAV\Server->getPropertiesIteratorForPath("photos/user/places", ["{http://nextcl ... "], 1)
 7. /var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php line 1647
    Sabre\DAV\Server->writeMultiStatus(["Sabre\\Xml\\Wr ... ]], ["Generator"], false)
 8. /var/www/html/3rdparty/sabre/dav/lib/DAV/CorePlugin.php line 346
    Sabre\DAV\Server->generateMultiStatus(["Generator"], false)
 9. /var/www/html/3rdparty/sabre/event/lib/WildcardEmitterTrait.php line 89
    Sabre\DAV\CorePlugin->httpPropFind(["Sabre\\HTTP\\Request"], ["Sabre\\HTTP\\Response"])
10. /var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php line 472
    Sabre\DAV\Server->emit("method:PROPFIND", [["Sabre\\HTTP\\ ... ]])
11. /var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php line 253
    Sabre\DAV\Server->invokeMethod(["Sabre\\HTTP\\Request"], ["Sabre\\HTTP\\Response"])
12. /var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php line 321
    Sabre\DAV\Server->start()
13. /var/www/html/apps/dav/lib/Server.php line 366
    Sabre\DAV\Server->exec()
14. /var/www/html/apps/dav/appinfo/v2/remote.php line 35
    OCA\DAV\Server->exec()
15. /var/www/html/remote.php line 171
    require_once("/var/www/html/a ... p")

Caused by:

Doctrine\DBAL\Exception\DriverException: An exception occurred while executing a query: SQLSTATE[42883]: Undefined function: 7 ERROR:  could not identify an equality operator for type json
LINE 1: SELECT DISTINCT "meta"."metadata" FROM "oc_file_metadata" "m...
                        ^ at <<closure>>

 0. /var/www/html/3rdparty/doctrine/dbal/src/Connection.php line 1814
    Doctrine\DBAL\Driver\API\PostgreSQL\ExceptionConverter->convert(["Doctrine\\DBAL ... "], ["Doctrine\\DBAL\\Query"])
 1. /var/www/html/3rdparty/doctrine/dbal/src/Connection.php line 1749
    Doctrine\DBAL\Connection->handleDriverException(["Doctrine\\DBAL ... "], ["Doctrine\\DBAL\\Query"])
 2. /var/www/html/3rdparty/doctrine/dbal/src/Connection.php line 1055
    Doctrine\DBAL\Connection->convertExceptionDuringQuery(["Doctrine\\DBAL ... "], "SELECT DISTINCT ... )", [3,6,"photos_place"], [1,1,2])
 3. /var/www/html/lib/private/DB/Connection.php line 264
    Doctrine\DBAL\Connection->executeQuery("SELECT DISTINCT ... )", [3,6,"photos_place"], [1,1,2], null)
 4. /var/www/html/3rdparty/doctrine/dbal/src/Query/QueryBuilder.php line 345
    OC\DB\Connection->executeQuery("SELECT DISTINCT ... )", [3,6,"photos_place"], [1,1,2])
 5. /var/www/html/lib/private/DB/QueryBuilder/QueryBuilder.php line 280
    Doctrine\DBAL\Query\QueryBuilder->execute()
 6. /var/www/html/lib/private/DB/QueryBuilder/QueryBuilder.php line 293
    OC\DB\QueryBuilder\QueryBuilder->execute()
 7. /var/www/html/apps/photos/lib/DB/Place/PlaceMapper.php line 62
    OC\DB\QueryBuilder\QueryBuilder->executeQuery()
 8. /var/www/html/apps/photos/lib/Sabre/Place/PlacesHome.php line 95
    OCA\Photos\DB\Place\PlaceMapper->findPlacesForUser("user")
 9. /var/www/html/3rdparty/sabre/dav/lib/DAV/Tree.php line 200
    OCA\Photos\Sabre\Place\PlacesHome->getChildren()
10. /var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php line 900
    Sabre\DAV\Tree->getChildren("photos/user/places")
11. /var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php line 982
    Sabre\DAV\Server->generatePathNodes(["Sabre\\DAV\\PropFind"], [["Sabre\\DAV\\P ... ]])
12. /var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php line 1662
    Sabre\DAV\Server->getPropertiesIteratorForPath("photos/user/places", ["{http://nextcl ... "], 1)
13. /var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php line 1647
    Sabre\DAV\Server->writeMultiStatus(["Sabre\\Xml\\Wr ... ]], ["Generator"], false)
14. /var/www/html/3rdparty/sabre/dav/lib/DAV/CorePlugin.php line 346
    Sabre\DAV\Server->generateMultiStatus(["Generator"], false)
15. /var/www/html/3rdparty/sabre/event/lib/WildcardEmitterTrait.php line 89
    Sabre\DAV\CorePlugin->httpPropFind(["Sabre\\HTTP\\Request"], ["Sabre\\HTTP\\Response"])
16. /var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php line 472
    Sabre\DAV\Server->emit("method:PROPFIND", [["Sabre\\HTTP\\ ... ]])
17. /var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php line 253
    Sabre\DAV\Server->invokeMethod(["Sabre\\HTTP\\Request"], ["Sabre\\HTTP\\Response"])
18. /var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php line 321
    Sabre\DAV\Server->start()
19. /var/www/html/apps/dav/lib/Server.php line 366
    Sabre\DAV\Server->exec()
20. /var/www/html/apps/dav/appinfo/v2/remote.php line 35
    OCA\DAV\Server->exec()
21. /var/www/html/remote.php line 171
    require_once("/var/www/html/a ... p")

Caused by:

Doctrine\DBAL\Driver\PDO\Exception: SQLSTATE[42883]: Undefined function: 7 ERROR:  could not identify an equality operator for type json
LINE 1: SELECT DISTINCT "meta"."metadata" FROM "oc_file_metadata" "m...
                        ^ at <<closure>>

 0. /var/www/html/3rdparty/doctrine/dbal/src/Driver/PDO/Statement.php line 103
    Doctrine\DBAL\Driver\PDO\Exception::new(["PDOException", ... ]])
 1. /var/www/html/3rdparty/doctrine/dbal/src/Connection.php line 1045
    Doctrine\DBAL\Driver\PDO\Statement->execute()
 2. /var/www/html/lib/private/DB/Connection.php line 264
    Doctrine\DBAL\Connection->executeQuery("SELECT DISTINCT ... )", [3,6,"photos_place"], [1,1,2], null)
 3. /var/www/html/3rdparty/doctrine/dbal/src/Query/QueryBuilder.php line 345
    OC\DB\Connection->executeQuery("SELECT DISTINCT ... )", [3,6,"photos_place"], [1,1,2])
 4. /var/www/html/lib/private/DB/QueryBuilder/QueryBuilder.php line 280
    Doctrine\DBAL\Query\QueryBuilder->execute()
 5. /var/www/html/lib/private/DB/QueryBuilder/QueryBuilder.php line 293
    OC\DB\QueryBuilder\QueryBuilder->execute()
 6. /var/www/html/apps/photos/lib/DB/Place/PlaceMapper.php line 62
    OC\DB\QueryBuilder\QueryBuilder->executeQuery()
 7. /var/www/html/apps/photos/lib/Sabre/Place/PlacesHome.php line 95
    OCA\Photos\DB\Place\PlaceMapper->findPlacesForUser("user")
 8. /var/www/html/3rdparty/sabre/dav/lib/DAV/Tree.php line 200
    OCA\Photos\Sabre\Place\PlacesHome->getChildren()
 9. /var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php line 900
    Sabre\DAV\Tree->getChildren("photos/user/places")
10. /var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php line 982
    Sabre\DAV\Server->generatePathNodes(["Sabre\\DAV\\PropFind"], [["Sabre\\DAV\\P ... ]])
11. /var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php line 1662
    Sabre\DAV\Server->getPropertiesIteratorForPath("photos/user/places", ["{http://nextcl ... "], 1)
12. /var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php line 1647
    Sabre\DAV\Server->writeMultiStatus(["Sabre\\Xml\\Wr ... ]], ["Generator"], false)
13. /var/www/html/3rdparty/sabre/dav/lib/DAV/CorePlugin.php line 346
    Sabre\DAV\Server->generateMultiStatus(["Generator"], false)
14. /var/www/html/3rdparty/sabre/event/lib/WildcardEmitterTrait.php line 89
    Sabre\DAV\CorePlugin->httpPropFind(["Sabre\\HTTP\\Request"], ["Sabre\\HTTP\\Response"])
15. /var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php line 472
    Sabre\DAV\Server->emit("method:PROPFIND", [["Sabre\\HTTP\\ ... ]])
16. /var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php line 253
    Sabre\DAV\Server->invokeMethod(["Sabre\\HTTP\\Request"], ["Sabre\\HTTP\\Response"])
17. /var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php line 321
    Sabre\DAV\Server->start()
18. /var/www/html/apps/dav/lib/Server.php line 366
    Sabre\DAV\Server->exec()
19. /var/www/html/apps/dav/appinfo/v2/remote.php line 35
    OCA\DAV\Server->exec()
20. /var/www/html/remote.php line 171
    require_once("/var/www/html/a ... p")

Caused by:

PDOException: SQLSTATE[42883]: Undefined function: 7 ERROR:  could not identify an equality operator for type json
LINE 1: SELECT DISTINCT "meta"."metadata" FROM "oc_file_metadata" "m...
                        ^ at <<closure>>

 0. /var/www/html/3rdparty/doctrine/dbal/src/Driver/PDO/Statement.php line 101
    PDOStatement->execute(null)
 1. /var/www/html/3rdparty/doctrine/dbal/src/Connection.php line 1045
    Doctrine\DBAL\Driver\PDO\Statement->execute()
 2. /var/www/html/lib/private/DB/Connection.php line 264
    Doctrine\DBAL\Connection->executeQuery("SELECT DISTINCT ... )", [3,6,"photos_place"], [1,1,2], null)
 3. /var/www/html/3rdparty/doctrine/dbal/src/Query/QueryBuilder.php line 345
    OC\DB\Connection->executeQuery("SELECT DISTINCT ... )", [3,6,"photos_place"], [1,1,2])
 4. /var/www/html/lib/private/DB/QueryBuilder/QueryBuilder.php line 280
    Doctrine\DBAL\Query\QueryBuilder->execute()
 5. /var/www/html/lib/private/DB/QueryBuilder/QueryBuilder.php line 293
    OC\DB\QueryBuilder\QueryBuilder->execute()
 6. /var/www/html/apps/photos/lib/DB/Place/PlaceMapper.php line 62
    OC\DB\QueryBuilder\QueryBuilder->executeQuery()
 7. /var/www/html/apps/photos/lib/Sabre/Place/PlacesHome.php line 95
    OCA\Photos\DB\Place\PlaceMapper->findPlacesForUser("user")
 8. /var/www/html/3rdparty/sabre/dav/lib/DAV/Tree.php line 200
    OCA\Photos\Sabre\Place\PlacesHome->getChildren()
 9. /var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php line 900
    Sabre\DAV\Tree->getChildren("photos/user/places")
10. /var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php line 982
    Sabre\DAV\Server->generatePathNodes(["Sabre\\DAV\\PropFind"], [["Sabre\\DAV\\P ... ]])
11. /var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php line 1662
    Sabre\DAV\Server->getPropertiesIteratorForPath("photos/user/places", ["{http://nextcl ... "], 1)
12. /var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php line 1647
    Sabre\DAV\Server->writeMultiStatus(["Sabre\\Xml\\Wr ... ]], ["Generator"], false)
13. /var/www/html/3rdparty/sabre/dav/lib/DAV/CorePlugin.php line 346
    Sabre\DAV\Server->generateMultiStatus(["Generator"], false)
14. /var/www/html/3rdparty/sabre/event/lib/WildcardEmitterTrait.php line 89
    Sabre\DAV\CorePlugin->httpPropFind(["Sabre\\HTTP\\Request"], ["Sabre\\HTTP\\Response"])
15. /var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php line 472
    Sabre\DAV\Server->emit("method:PROPFIND", [["Sabre\\HTTP\\ ... ]])
16. /var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php line 253
    Sabre\DAV\Server->invokeMethod(["Sabre\\HTTP\\Request"], ["Sabre\\HTTP\\Response"])
17. /var/www/html/3rdparty/sabre/dav/lib/DAV/Server.php line 321
    Sabre\DAV\Server->start()
18. /var/www/html/apps/dav/lib/Server.php line 366
    Sabre\DAV\Server->exec()
19. /var/www/html/apps/dav/appinfo/v2/remote.php line 35
    OCA\DAV\Server->exec()
20. /var/www/html/remote.php line 171
    require_once("/var/www/html/a ... p")

PROPFIND /remote.php/dav/photos/user/places/ 
```

**Comment by davidcroda** [comment]
I think this could be related to the fact that the photos app seems to write the place name as plaintext into a `json` type field. I had a problem when converting from a sqlite database to a postgres database where it was failing due to the string "Dresher" being present in the `ocfile_metadata.metadata` json field. Sqlite does not validate type data but postgres does.

I think in the [`setPlaceForFile`](https://github.com/nextcloud/photos/blob/66cd90f11a206421d9cd56315cf066795ed71b3c/lib/DB/Place/PlaceMapper.php#L175) function `place` should be set into a proper json object, ie. `{"place": $place}` when being inserted into the table. I just started digging into this so there are other queries I'm sure which would need to correctly query on the now valid json as well.

**Comment by kpaaro** [comment]
I get this exception as but i am using postgres. The error reads 
```
2023-04-20 17:34:13.589 EEST [10727] ERROR:  could not identify an equality operator for type json at character 17
2023-04-20 17:34:13.589 EEST [10727] STATEMENT:  SELECT DISTINCT "meta"."metadata" FROM "oc_file_metadata" "meta" INNER JOIN "oc_filecache" "file" ON "file"."fileid" = "meta"."id" WHERE ("file"."storage" = $1) AND ("file"."mimepart" = $2) AND ("meta"."group_name" = $3)
```

**Comment by joshtrichards** [comment]
Fixed in nextcloud/server#37146 and #1692. Released in NC27 and backported to v26.0.1.

@vanantwerp - Please confirm you can no longer reproduce this behavior and close this Issue out. Thanks!

**Comment by vanantwerp** [comment]
@joshtrichards apologies but I've since switched to the AIO and am unable to go back to the environment I originally noticed the issue in. Thank you for your work on the fix!
